From d4c9494eb545ff7fffbf84504186dff550ba0602 Mon Sep 17 00:00:00 2001
From: Tom de Geus <tom@geus.me>
Date: Wed, 20 Jan 2021 21:12:18 +0100
Subject: [PATCH 2/2] Making target detection on Mac more robust

---
 include/internal/catch_platform.h | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/include/internal/catch_platform.h b/include/internal/catch_platform.h
index e14cd332..82c5e144 100644
--- a/include/internal/catch_platform.h
+++ b/include/internal/catch_platform.h
@@ -9,13 +9,16 @@
 #ifndef TWOBLUECUBES_CATCH_PLATFORM_H_INCLUDED
 #define TWOBLUECUBES_CATCH_PLATFORM_H_INCLUDED
 
+// See e.g.:
+// https://opensource.apple.com/source/CarbonHeaders/CarbonHeaders-18.1/TargetConditionals.h.auto.html
 #ifdef __APPLE__
-# include <TargetConditionals.h>
-# if TARGET_OS_OSX == 1
-#  define CATCH_PLATFORM_MAC
-# elif TARGET_OS_IPHONE == 1
-#  define CATCH_PLATFORM_IPHONE
-# endif
+#  include <TargetConditionals.h>
+#  if (defined(TARGET_OS_OSX) && TARGET_OS_OSX == 1) || \
+      (defined(TARGET_OS_MAC) && TARGET_OS_MAC == 1)
+#    define CATCH_PLATFORM_MAC
+#  elif (defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE == 1)
+#    define CATCH_PLATFORM_IPHONE
+#  endif
 
 #elif defined(linux) || defined(__linux) || defined(__linux__)
 #  define CATCH_PLATFORM_LINUX
-- 
2.24.3 (Apple Git-128)

